From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Sun, 05 Jul 2026 05:20:13 +0000
Subject: [PATCH] Gecko - IronFox Settings: XPInstall

Provides support for a UI setting to enable/disable the installation of add-ons.

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/browser/components/enterprisepolicies/Policies.sys.mjs b/browser/components/enterprisepolicies/Policies.sys.mjs
index e612b78173..f04524a16a 100644
--- a/browser/components/enterprisepolicies/Policies.sys.mjs
+++ b/browser/components/enterprisepolicies/Policies.sys.mjs
@@ -31,6 +31,7 @@ ChromeUtils.defineESModuleGetters(lazy, {
   SearchService: "moz-src:///toolkit/components/search/SearchService.sys.mjs",
   QuickSuggest: "moz-src:///browser/components/urlbar/QuickSuggest.sys.mjs",
   WebsiteFilter: "resource:///modules/policies/WebsiteFilter.sys.mjs",
+  IFPrefUtils:   "moz-src:///ironfox/utils/IFPrefUtils.sys.mjs",
 });
 
 const PREF_LOGLEVEL = "browser.policies.loglevel";
@@ -1954,8 +1955,8 @@ export var Policies = {
         addAllowDenyPermissions("install", param.Allow, null);
       }
       if ("Default" in param) {
-        setAndLockPref("xpinstall.enabled", param.Default);
-        if (!param.Default) {
+//        setAndLockPref("xpinstall.enabled", param.Default);
+        if (!lazy.IFPrefUtils.getBoolPref("xpinstall.enabled")) {
           manager.disallowFeature("installTemporaryAddon");
           setAndLockPref(
             "browser.newtabpage.activity-stream.asrouter.userprefs.cfr.addons",
-- 
